tests/test_morphsqueeze.py

```diff
@@ -170,3 +170,110 @@ def test_morphsqueeze_extrapolate(user_filesystem, squeeze_coeffs, wmsg_gen):
     )
     with pytest.warns(UserWarning, match=expected_wmsg):
         single_morph(parser, opts, pargs, stdout_flag=False)
+
+
+def test_non_unique_grid():
+    squeeze_coeffs = {"a0": 0.01, "a1": 0.01, "a2": -0.1}
+    x_grid = np.linspace(0, 10, 101)
+
+    coeffs = [squeeze_coeffs[f"a{i}"] for i in range(len(squeeze_coeffs))]
+    squeeze_polynomial = Polynomial(coeffs)
+    x_morph = x_grid + squeeze_polynomial(x_grid)
+    x_gradient = np.diff(x_morph)
+    x_gradient_sign = np.sign(x_gradient)
+    # non-strictly increasing means the gradient becomes 0 or negative
+    assert not np.all(x_gradient_sign > 0)
+
+    x_target = np.linspace(min(x_morph), max(x_morph), len(x_morph))
+    y_target = np.sin(x_target)
+
+    y_morph = np.sin(x_morph)
+    # apply no squeeze, but the morph should sort the function
+    _, table = morphpy.morph_arrays(
+        np.array([x_morph, y_morph]).T,
+        np.array([x_target, y_target]).T,
+        squeeze=[0, 0, 0],
+        apply=True,
+    )
+    x_refined, _ = table[:, 0], table[:, 1]
+
+    # grid should be properly sorted
+    assert np.allclose(x_refined, x_target)
+    # note that the function itself may be distorted
+
+
+@pytest.mark.parametrize(
+    "squeeze_coeffs, x_morph",
+    [
+        # The following squeezes make the function non-monotonic
+        ({"a0": -1, "a1": -1, "a2": 2}, np.linspace(-1, 1, 101)),
+        (
+            {"a0": -1, "a1": -1, "a2": 0, "a3": 0, "a4": 2},
+            np.linspace(-1, 1, 101),
+        ),
+    ],
+)
+def test_squeeze_warnings(user_filesystem, squeeze_coeffs, x_morph):
+    # call in .py
+    x_target = x_morph
+    y_target = np.sin(x_target)
+    coeffs = [squeeze_coeffs[f"a{i}"] for i in range(len(squeeze_coeffs))]
```

Comment:
is there a reason you don't just give as "squeeze_coeffs" inputs `[-1, -1, 0, 0, 2]` and remove this line?  I think it would be easier to read.  Less cognifive overload for the poor reviewer....